source/security/authentication.txt

```diff
@@ -0,0 +1,403 @@
+identity provider, the {+php-library+} can make an ``AssumeRoleWithWebIdentity`` request
+to exchange the OIDC token for temporary AWS credentials for your application. 
+
+To authenticate with temporary AWS IAM credentials returned by an
+``AssumeRoleWithWebIdentity`` request, ensure that the AWS config file exists in your
+environment and is configured correctly. To learn how to create and configure
+an AWS config file, see `Configuration <https://docs.aws.amazon.com/sdkref/latest/guide/creds-config-files.html>`__
+in the AWS documentation.
+
+After you configure your environment for an ``AssumeRoleWithWebIdentity`` request,
+set the ``authMechanism`` connection option to ``'MONGODB-AWS'``.
+To view an example that sets the ``authMechanism`` option, see the :ref:`authMechanism example
+<php-mongodb-aws-env-example>` on this page.
+
+.. tip::
+
+   For more information about using an ``AssumeRoleWithWebIdentity`` request to
+   authenticate your application, see the following AWS documentation:
+
+   - `Authenticating users for your cluster from an OpenID Connect identity provider
+     <https://docs.aws.amazon.com/eks/latest/userguide/authenticate-oidc-identity-provider.html>`__
+   - `AssumeRoleWithWebIdentity
+     <https://docs.aws.amazon.com/STS/latest/APIReference/API_AssumeRoleWithWebIdentity.html>`__
+
+.. _php-mongodb-aws-ecs:
+
+ECS Metadata
+~~~~~~~~~~~~
+
+If your application runs in an Elastic Container Service (ECS) container,
+the {+php-library+} can automatically retrieve temporary AWS credentials from an
+ECS endpoint. To do so, specify the URI of the ECS endpoint in an environment variable called
+``AWS_CONTAINER_CREDENTIALS_RELATIVE_URI``, as shown in the following example:
+
+.. code-block:: sh
+
+   export AWS_CONTAINER_CREDENTIALS_RELATIVE_URI=<URI of the ECS endpoint>
+
+After you set the environment variable, set the ``authMechanism``
+connection option to ``'MONGODB-AWS'``. To view an example that sets the
+``authMechanism`` option, see the :ref:`authMechanism example
+<php-mongodb-aws-env-example>` on this page.
+
+.. _php-mongo-aws-ec2:
+
+EC2 Instance Metadata
+~~~~~~~~~~~~~~~~~~~~~
+
+The {+driver-short+} can automatically retrieve temporary AWS credentials from an
+Amazon Elastic Cloud Compute (EC2) instance. To use temporary credentials from
+within an EC2 instance, set the ``authMechanism`` connection option to ``'MONGODB-AWS'``.
+To view an example that sets the ``authMechanism`` option, see the :ref:`authMechanism example
+<php-mongodb-aws-env-example>` on this page.
+
+.. note::
+
+   If you set any of the environment variables from the preceding AWS authentication
+   methods, the {+php-library+} attempts to retrieve credentials by using those
+   methods before attempting to retrieve them from an EC2 instance. To attempt
+   to retrieve credentials only from an EC2 instance, ensure that the
+   environment variables are not set.
```

Comment:
S: confusing use of "method"

```suggestion
   The {+php-library+} retrieves credentials from an EC2 instance only if the
   environment variables described earlier in this section are not set.
```

Consider linking to that earlier section, too